engine/src/main/java/org/entando/entando/aps/system/services/group/GroupService.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2015-Present Entando Inc. (http://www.entando.com) All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+package org.entando.entando.aps.system.services.group;
+
+import java.util.Arrays;
+import java.util.List;
+
+import com.agiletec.aps.system.common.model.dao.SearcherDaoPaginatedResult;
+import com.agiletec.aps.system.exception.ApsSystemException;
+import com.agiletec.aps.system.services.group.Group;
+import com.agiletec.aps.system.services.group.IGroupManager;
+import org.entando.entando.aps.system.exception.RestRourceNotFoundException;
+import org.entando.entando.aps.system.exception.RestServerError;
+import org.entando.entando.aps.system.services.IDtoBuilder;
+import org.entando.entando.aps.system.services.group.model.GroupDto;
+import org.entando.entando.web.common.model.PagedMetadata;
+import org.entando.entando.web.common.model.RestListRequest;
+import org.entando.entando.web.group.model.GroupRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class GroupService implements IGroupService {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private IGroupManager groupManager;
+
+    @Autowired
+    private IDtoBuilder<Group, GroupDto> dtoBuilder;
+
+
+    protected IGroupManager getGroupManager() {
+        return groupManager;
+    }
+
+    public void setGroupManager(IGroupManager groupManager) {
+        this.groupManager = groupManager;
+    }
+
+    protected IDtoBuilder<Group, GroupDto> getDtoBuilder() {
+        return dtoBuilder;
+    }
+
+    public void setDtoBuilder(IDtoBuilder<Group, GroupDto> dtoBuilder) {
+        this.dtoBuilder = dtoBuilder;
+    }
+
+    @Override
+    public PagedMetadata<GroupDto> getGroups(RestListRequest restListReq) {
+        try {
+
+            Arrays.stream(restListReq.getFieldSearchFilters())
```

Comment:
Lets add a comment here on what this is doing. Especially since this is the reference implementation.